java/src/org/boris/pecoff4j/io/ResourceParser.java

```diff
@@ -321,7 +321,11 @@ public static IconDirectory readIconDirectory(IDataReader dr)
 
 	private static int alignDataReader(IDataReader dr) throws IOException {
 		int off = (4 - (dr.getPosition() % 4)) % 4;
-		dr.skipBytes(off);
+		for (int i = 0; i < off; i++) {
+			if (!dr.hasMore())
+				break;
+			dr.skipBytes(1);
+		}
```

Comment:
Maybe catching `EOFException` and adding a comment would look cleaner? Something like:

```
try {
  dr.skipBytes(off);
} catch (EOFException) {
  // no need to align at the end of the file
}
```

Honestly, just a comment would be enough. It took me a few reads to understand why this was OK.

Reply:
reasonable